Add GA-only conformance job to master-blocking testgrid

liggitt opened the pull request:
Part of https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/20191023-conformance-without-beta.md

The periodic GA-only conformance job ensures that no new beta APIs or features are introduced as requirements to pass conformance tests. In the four days this job has existed, it has already caught one breaking change.

@kubernetes/sig-release 
for ack of addition to release-blocking dashboard

@kubernetes/sig-architecture-pr-reviews 
for surfacing of conformance results as part of release criteria

liggitt commented:
/sig release

dims commented:
/assign @cblecker 
/approve
/lgtm

cblecker commented on config/jobs/kubernetes/sig-testing/conformance-e2e.yaml:
```diff
@@ -271,6 +271,6 @@ periodics:
     testgrid-num-failures-to-alert: '2'
     testgrid-alert-stale-results-hours: '48'
-    testgrid-dashboards: conformance-all, conformance-kind, sig-testing-kind
+    testgrid-dashboards: conformance-all, conformance-kind, sig-testing-kind, sig-release-master-blocking
```
```
    config_test.go:480: dashboard_tab sig-release-master-blocking/conformance-ga-only is release-blocking: - Must have alert_options
```

liggitt replied:
I thought it did...

```
  annotations:
    testgrid-num-failures-to-alert: '2'
    testgrid-alert-stale-results-hours: '48'
    testgrid-dashboards: conformance-all, conformance-kind, sig-testing-kind, sig-release-master-blocking
    testgrid-tab-name: conformance-ga-only
    testgrid-alert-email: liggitt@google.com

```

cblecker replied:
I think what it's missing actually is a description.

liggitt replied:
oh, ok... added a description

justaugustus commented on config/jobs/kubernetes/sig-testing/conformance-e2e.yaml:
```diff
@@ -271,6 +271,6 @@ periodics:
-    testgrid-dashboards: conformance-all, conformance-kind, sig-testing-kind
+    testgrid-dashboards: conformance-all, conformance-kind, sig-testing-kind, sig-release-master-blocking
     testgrid-tab-name: conformance-ga-only
     testgrid-alert-email: liggitt@google.com
```
Can we have a non-human alert group here?
If it's on master-blocking kubernetes-release-team@googlegroups.com ideally is also one of the alerted parties.

liggitt replied:
changed to sig-release and sig-architecture (in lieu of a conformance-specific list)

justaugustus commented on config/jobs/kubernetes/sig-testing/conformance-e2e.yaml:
```diff
@@ -271,6 +271,7 @@ periodics:
+    testgrid-dashboards: conformance-all, conformance-kind, sig-testing-kind, sig-release-master-blocking
     testgrid-tab-name: conformance-ga-only
     testgrid-alert-email: liggitt@google.com
+    description: "conformance tests run against a master kind cluster with only GA APIs and features enabled"
```
I _believe_ this description needs to start with: `OWNER: <owner> blah blah blah`
We should look at making these configurations more clear moving into 1.19.

liggitt replied:
added

liggitt commented:
the thing the alert test was looking for required the informing/blocking dashboard to be first in the list... fixed that and updated the test failure text to make that clear

justaugustus commented:
You rock, @liggitt!

For SIG Release:
/lgtm
/approve

cblecker commented:
/approve
